nova/virt/vmwareapi/driver.py

```diff
@@ -876,6 +877,109 @@ def _server_group_sync_loop(self, compute_host):
 
             time.sleep(CONF.vmware.server_group_sync_loop_spacing)
 
+    def _custom_traits_sync_loop(self, compute_host):
+        """Get the traits from Placement and sync them down into the cluster's
+        custom attributes.
+
+        Every trait needs to become an CustomFieldDef through the
+        customFieldManager. These CustomFieldDef are visible on all objects of
+        the type they're defined for - just without value. If a trait is active
+        for an object, we set the value to "true". Otherwise, we keep it empty.
+        """
+        context = nova_context.get_admin_context()
+
+        # in addition to all "CUSTOM_"-prefixed traits, we also sync these
+        # traits into the cluster
+        EXTRA_TRAITS = set(['COMPUTE_STATUS_DISABLED'])
+
+        PREFIX = constants.CUSTOM_ATTRIBUTES_TRAITS_PREFIX
+
+        while CONF.vmware.custom_traits_sync_loop_spacing >= 0:
+            LOG.debug('Starting custom traits sync-loop')
+            try:
+                placement_client = self.virtapi.reportclient
+                # NOTE(jkulik): We need to get all traits, because while our
+                # nova-compute might not need a trait, other nova-compute in
+                # the same vCenter might. To not delete CustomFieldDef that's
+                # actually in use by others, we keep all traits in the vCenter.
+                trait_fields_names = set(
+                    f"{PREFIX}{t}"
+                    for t in placement_client.get_traits(context)
+                    if t.startswith('CUSTOM_') or t in EXTRA_TRAITS)
+
+                custom_field = nova_vim_util.CustomField(self._session)
+                obj_type = 'ClusterComputeResource'
+                # Fetch the existing CustomFieldDef
+                vc_fields_names_by_key = {
+                    f.key: f.name
+                    for f in custom_field.get_by_obj_type(obj_type)
+                    if f.name.startswith(PREFIX)}
```

Comment:
This lists all the `openstack.nova.traits.` custom attributes available in the vCenter. It will find the `EXTRA_TRAITS` in there in addition to all the `CUSTOM_`-prefixed ones. So no, we do not need `EXTRA_TRAITS` here.